modules/programs/git.nix

```
@@ -130,10 +130,19 @@ let
           </citerefentry>.
         '';
       };
-    };
 
-    config.path = mkIf (config.contents != { })
-      (mkDefault (pkgs.writeText "contents" (gitToIni config.contents)));
+      contentSuffix = mkOption {
+        type = types.str;
+        default = "gitconfig";
+        description = ''
+          Nix store name for the git configuration text file,
+          when generating the configuration text from nix options.
+        '';
+
+      };
+    };
+    config.path = mkIf (config.contents != { }) (mkDefault
+      (pkgs.writeText config.contentSuffix (gitToIni config.contents)));
```

The filename should go through `hm.strings.storeFileName` to strip invalid characters.
```suggestion
      (pkgs.writeText (hm.strings.storeFileName config.contentSuffix) (gitToIni config.contents)));
```
https://github.com/nix-community/home-manager/blob/3d3bbdfe955383033fd17c28d2b1e93d48515d7d/modules/lib/strings.nix#L8-L22

---

nice catch, will add a test for the case